polygeist/tools/cgeist/Test/Verification/sycl/sycl_extern_func.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) Intel
+
+//===--- sycl_extern_func.cpp ---------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// RUN: sycl-clang.py %s -S | FileCheck %s --check-prefix=CHECK-MLIR
+// RUN: sycl-clang.py %s -S -emit-llvm | FileCheck %s --check-prefix=CHECK-LLVM
+
+#include <sycl/sycl.hpp>
+
+// CHECK-MLIR-LABEL: func.func @cons_5() attributes {llvm.cconv = #llvm.cconv<spir_funccc>, llvm.linkage = #llvm.linkage<external>, passthrough = ["norecurse", "nounwind", "convergent", "mustprogress"]} {
```

Comment:
Would be good if test case could check that there is a GPUModuleOp containing the cons_5 function definition and the definition for the kernel corresponding to the SYCL single_task construct.

Reply:
True, pushing...